File: user-auditor/src/main/java/com/utmstack/userauditor/model/event/Event.java
package com.utmstack.userauditor.model.event;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.Getter;

import java.util.Map;


@Data
public class Event {
    private String id;
    @JsonProperty("@timestamp")
    private String timestamp;
    private String deviceTime;
    private String dataType;
    private String dataSource;
    private String tenantId;
    private String tenantName;
    private String raw;
    private Map<String, Object> log;
    private Side target;
    private Side origin;
    private String protocol;
    private String connectionStatus;
    private int statusCode;
    private String actionResult;
    private String action;
    private String command;
    private String severity;
}

File: frontend-legacy/src/app/shared/types/event/event.ts
export interface Geolocation {
  country: string;
  city: string;
  latitude: number;
  longitude: number;
  asn: number;
  aso: string;
  countryCode: string;
  accuracy: number;
}

export interface Side {
  bytesSent: number;
  bytesReceived: number;
  packagesSent: number;
  packagesReceived: number;
  connections: number;
  usedCpuPercent: number;
  usedMemPercent: number;
  totalCpuUnits: number;
  totalMem: number;
  ip: string;
  host: string;
  user: string;
  group: string;
  port: number;
  domain: string;
  fqdn: string;
  mac: string;
  process: string;
  geolocation: Geolocation;
  file: string;
  path: string;
  hash: string;
  url: string;
  email: string;
}

export interface Event {
  id: string;
  timestamp: string;
  deviceTime: string;
  dataType: string;
  dataSource: string;
  tenantId: string;
  tenantName: string;
  raw: string;
  log: { [key: string]: any };
  target: Side;
  origin: Side;
  protocol: string;
  connectionStatus: string;
  statusCode: number;
  actionResult: string;
  action: string;
  command: string;
  severity: string;
}
